icepyx/core/query.py

```diff
@@ -854,58 +849,55 @@ def show_custom_options(self, dictview=False):
+        Authentication is completed using the [earthaccess library](https://nsidc.github.io/earthaccess/).
+        Methods for authenticating are:
+            1. Storing credentials as environment variables ($EARTHDATA_LOGIN and $EARTHDATA_PASSWORD)
+            2. Entering credentials interactively
+            3. Storing credentials in a .netrc file (not recommended for security reasons)
+        More details on using these methods is available in the [earthaccess documentation](https://nsidc.github.io/earthaccess/tutorials/restricted-datasets/#auth).
+        The input parameters listed here are provided for backwards compatibility;
+        before earthaccess existed, icepyx handled authentication and required these inputs.
 
         Parameters
         ----------
-        uid : string
-            Earthdata login user ID
-        email : string
-            Email address. NSIDC will automatically send you emails about the status of your order.
+        uid : string, default None
+            Deprecated keyword for Earthdata login user ID.
+        email : string, default None
+            Deprecated keyword for backwards compatibility.
         s3token : boolean, default False
-            Generate AWS s3 ICESat-2 data access credentials
-
-        See Also
-        --------
-        Earthdata.Earthdata
+            Deprecated keyword to generate AWS s3 ICESat-2 data access credentials
+        kwargs : key:value pairs
+            Keyword arguments to be passed into earthaccess.login().
 
         Examples
         --------
         >>> reg_a = ipx.Query('ATL06',[-55, 68, -48, 71],['2019-02-20','2019-02-28']) # doctest: +SKIP
-        >>> reg_a.earthdata_login(user_id,user_email) # doctest: +SKIP
-        Earthdata Login password:  ········
-        """
-
-        if s3token == False:
-            capability_url = f"https://n5eil02u.ecs.nsidc.org/egi/capabilities/{self.product}.{self._version}.xml"
-        elif s3token == True:
+        >>> reg_a.earthdata_login() # doctest: +SKIP
+        Enter your Earthdata Login username: ___________________
 
-            def is_ec2():
-                import socket
+        EARTHDATA_USERNAME and EARTHDATA_PASSWORD are not set in the current environment, try setting them or use a different strategy (netrc, interactive)
+        No .netrc found in /Users/username
 
-                try:
-                    socket.gethostbyname("instance-data")
-                    return True
-                except socket.gaierror:
-                    return False
-
-            # loosely check for AWS login capability without web request
-            assert (
-                is_ec2() == True
-            ), "You must be working from a valid AWS instance to use s3 data access"
-            capability_url = "https://data.nsidc.earthdatacloud.nasa.gov/s3credentials"
+        """
 
-        self._session = Earthdata(uid, email, capability_url).login()
+        auth = earthaccess.login(**kwargs)
```

Comment:
These are not strictly used by earthaccess but I assume is to keep the API the same for Icepyx right?

Reply:
the **kwargs? I put this in so that if a savvy user wanted to pass `persist=True` or one of the other earthaccess login keywords they could.